### PR TITLE
Add conditionable trait to `Panel` class

### PR DESCRIPTION
**Pull request by Summonshr**
- use Conditionable Trait.

use case - 

```php
$panel->when(!app()->runningInConsole(), function ($panel) {
    return $panel
        ->viteTheme('resources/css/filament/admin/theme.css');
})
```

### CODE REVIEW

File: packages/panels/src/Panel.php [hunk 1]
```diff
@@ -33,7 +33,8 @@ class Panel extends Component
     use Panel\Concerns\HasTheme;
     use Panel\Concerns\HasTopNavigation;
     use Panel\Concerns\HasUserMenu;
-
+    use \Illuminate\Support\Traits\Conditionable;
```
Comment (zepfietje) [inline packages/panels/src/Panel.php]: Please import the namespace.

File: packages/panels/src/Panel.php [hunk 2]
```diff
@@ -33,7 +34,7 @@ class Panel extends Component
     use Panel\Concerns\HasTheme;
     use Panel\Concerns\HasTopNavigation;
     use Panel\Concerns\HasUserMenu;
-    use \Illuminate\Support\Traits\Conditionable;
+    use Conditionable;
```
Comment (zepfietje) [inline packages/panels/src/Panel.php]: Move it to the top to be sorted alphabetically please. 🙂

File: packages/panels/src/Panel.php (hunk 1 above)
Comment (Summonshr) [reply packages/panels/src/Panel.php]: Imported.

**Comment by zepfietje** [comment]
Maybe we should add `Conditionable` to `Filament\Support\Components` instead, @danharrin?

**Comment by Summonshr** [comment]
> Maybe we should add `Conditionable` to `Filament\Support\Components` instead, @danharrin?

I tried this actually. Not sure what did not work.

**Comment by Summonshr** [comment]
> Maybe we should add `Conditionable` to `Filament\Support\Components` instead, @danharrin?

It conflicts with CanBeHidden Traits.

`Declaration of Filament\Forms\Components\Concerns\CanBeHidden::when(Closure|bool $condition = true): static must be compatible with Filament\Support\Components\Component::when($value = null, ?callable $callback = null, ?callable $default = null)`

**Comment by zepfietje** [comment]
Ah, I see. Our `when()` method is an alias of `visible()`, so we might remove `when()` in favor of adding the `Conditionable` trait. This would be a breaking change though, so would have to be implemented before releasing v3 stable, @danharrin.

**Comment by danharrin** [comment]
I'm ok with that, we can remove the old `when()` methods

**Comment by danharrin** [comment]
Mark this as ready when you've done that @Summonshr

**Comment by Summonshr** [comment]
> Mark this as ready when you've done that @Summonshr

Done

File: packages/panels/src/Panel.php [hunk 3]
```diff
@@ -33,7 +33,7 @@ class Panel extends Component
     use Panel\Concerns\HasTheme;
     use Panel\Concerns\HasTopNavigation;
     use Panel\Concerns\HasUserMenu;
-
+
```
Comment (zepfietje) [inline packages/panels/src/Panel.php]:
```suggestion

```

**Comment by zepfietje** [comment]
Thanks

**Pull request merged**